single-line-diagram-core/src/main/java/com/powsybl/sld/util/TopologicalStyleProvider.java

```diff
@@ -149,6 +149,21 @@ private RGBColor getSmartNodeColor(VoltageLevelInfos voltageLevelInfos, Node nod
         return null;
     }
 
+    @Override
+    protected String getEdgeColor(Node node1, Node node2) {
+        if (node1.getType() == NodeType.SWITCH && node1.isOpen()) {
+            return node2.getVoltageLevelInfos() != null ? getNodeColor(node2.getVoltageLevelInfos(), node2) : null;
+        }
+        if (node2.getType() == NodeType.SWITCH && node2.isOpen()) {
+            return node1.getVoltageLevelInfos() != null ? getNodeColor(node1.getVoltageLevelInfos(), node1) : null;
+        }
+        if (node1.getVoltageLevelInfos() != null) {
```

Comment:
Maybe here just a call to super.getEdgeColor(...) ?

Reply:
Yes, corrected.